src/internal/connector/onedrive/collection.go

```diff
@@ -53,6 +53,13 @@ var (
 	_ data.StreamModTime    = &metadataItem{}
 )
 
+type SharingMode int
+
+const (
+	SharingModeCustom = SharingMode(iota)
+	SharingModeInherited
+)
```

Comment:
- I don't have a strong opinion on `string` vs `int`. I choose `int` mainly because it is lesser bytes (plus iota felt more like using enums)
- I don't mind switching to strings. As you mentioned, they are much more readable. **Although, any opinion on using `TitleCase` instead of `snake_case`?**
- About `SharingModeUnknown`, I initially had it, but removed as treating older backups without permissions as `SharingModeCustom` is the "right" choice anyways. Plus, the "omitempty" in the struct will drop `SharingModeCustom` even as of now.
- Another possibility is to just convert it to a boolean as there are only two options. I created it as an "enum" as we had more options(`SharingModeEmpty`) initially. I don't think we will need more options here, but I just didn't want to take a risk.

Reply:
>I don't think we will need more options here, but I just didn't want to take a risk

seems reasonable

>They tend to be better for readability

This is a limitation of golang, not the approach of using ints for enums. Also Abin has a good point that the number of bytes can matter. We haven't really been worrying about this too much to this point (after all, we're using JSON encoding for a lot of things...), but in storage and networking it's pretty common to try to minimize the number of bytes that need to be stored/sent as much as possible. That's why enums in those codebases often end up as integers instead of strings

Overall, I'm hesitant to switch persisted enum values to strings (I'm a bit less worried about non-persisted enums as we can change those however much we want since they don't affect backwards compatibility). Int enums have nice default values that we can associate with things and add meaning to. For example, having 0 be the "Unknown" (or whatever default we need it to be) value works well to catch things that weren't populated to begin with or things that weren't set explicitly. Strings don't have the equivalent as having a "meaningful" name for it would imply `const UnknownType = "unknown_type"` which is different from `""` that will be returned for unpopulated values. Having `const UnknownType = ""` or `const LegacyType = ""` is equally unhelpful if we assume we're just printing these values as an empty string in the log is unhelpful; putting us right back in the situation we said string enums would help us avoid

Reply:
> This is a limitation of golang, not the approach of using ints for enums.

I too share a similar opinion.

I think I'll stick with int for now, plus we can always write `Stringer`s if necessary.